lib/url.js

```diff
@@ -948,3 +948,67 @@ function spliceOne(list, index) {
     list[i] = list[k];
   list.pop();
 }
+
+function encodeAuth(str) {
```

Comment:
I know this originally came from `querystring.escape()`, but that function's implementation has recently improved performance-wise and we should probably incorporate those changes here (specifically the ASCII value checking -- it uses a lookup table now).